src/libsync/propagateuploadng.cpp

```diff
@@ -455,6 +455,18 @@ void PropagateUploadFileNG::slotMoveJobFinished()
         commonErrorHandling(job);
         return;
     }
+
+    if (_item->_httpErrorCode == 202) {
+        QString path = QString::fromUtf8(job->reply()->rawHeader("OC-JobStatus-Location"));
+        if (path.isEmpty()) {
+            done(SyncFileItem::NormalError, tr("Poll URL missing"));
+            return;
+        }
+        _finished = true;
+        startPollJob(path);
```

Comment:
How does the overall sync timeout work in the presence of polls?

Reply:
The timeouts are per jobs.  A particular PollJob will timeout after the normal timeout time if there is not traffic on that connection. In this case, it will be re-tried again until it succeed.

Reply:
I meant to confirm that if the server needs 30 minutes to finish the upload then the sync will only finish once that last poll is done. (as it should be for 2.5.1)